.vvv/lib/config.rb

```diff
@@ -0,0 +1,160 @@
+# frozen_string_literal: true
+
+module VVV
+  class Config
+    GITHUB_USER_URL    = 'https://github.com/Varying-Vagrant-Vagrants/'
+    UTILITIES_REPO_URL = "#{GITHUB_USER_URL}vvv-utilities.git"
+    DASHBOARD_REPO_URL = "#{GITHUB_USER_URL}dashboard.git"
+    PRIVATE_NETWORK_IP = '192.168.56.4'
+
+    CONFIG_FILE = File.join(VVV::Info.vagrant_dir, 'config/config.yml').freeze
+
+    require 'yaml'
+
+    def initialize
+      @config = YAML.load_file(CONFIG_FILE)
+      lint
+    end
+
+    def values
+      @config
+    end
+
+    def lint
+      # If 'hosts' isn't an array, redefine it as such
+      @config['hosts'] = ['vvv.test'] unless @config['hosts'].is_a? Array
+
+      @config['sites'].each do |site, args|
+        # If the site's value is a string, treat it as the repo value
+        if @config['sites'][site].is_a? String
+          @config['sites'][site] = { repo: args }
+        end
+
+        # If the site's args aren't defined as a Hash already,
+        # redefine it as such.
+        @config['sites'][site] = {} unless @config['sites'][site].is_a? Hash
+
+        merge_site_defaults(site)
+        process_hosts_for_site(site)
+      end
+      process_dashboard
+      process_utilities
+      process_utility_sources
+      process_extensions
+      process_extension_sources
```

Comment:
these are the wrong way around
```suggestion
      process_extension_sources
      process_utility_sources
      process_extensions
      process_utilities
```

Reply:
though it is just linting so this isn't as important as it looked when I read it